From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shelley Vohr <shelley.vohr@gmail.com>
Date: Wed, 21 Aug 2019 12:43:26 -0700
Subject: Add OPENSSL_IS_BORINGSSL guard to oaep_hash check

Adds an ifndef guard to the oaep_hash nullptr check in
PublicKeyCipher::Cipher, since BORINGSSL does not expose either
EVP_PKEY_OP_TYPE_CRYPT or EVP_PKEY_CTRL_RSA_OAEP_MD.

diff --git a/src/node_crypto.cc b/src/node_crypto.cc
index 59b60f1aea11666a7393831f8a465c247b66b009..1b44572c3a126c1ba3f610dbefb226acb34d246d 100644
--- a/src/node_crypto.cc
+++ b/src/node_crypto.cc
@@ -5156,11 +5156,15 @@ bool PublicKeyCipher::Cipher(Environment* env,
   if (EVP_PKEY_CTX_set_rsa_padding(ctx.get(), padding) <= 0)
     return false;
 
+#ifdef OPENSSL_IS_BORINGSSL
+  env->ThrowError("oaepHash option not supported in BoringSSL.");
+#else
   if (oaep_hash != nullptr) {
     if (!EVP_PKEY_CTX_md(ctx.get(), EVP_PKEY_OP_TYPE_CRYPT,
                          EVP_PKEY_CTRL_RSA_OAEP_MD, oaep_hash))
       return false;
   }
+#endif
 
   size_t out_len = 0;
   if (EVP_PKEY_cipher(ctx.get(), nullptr, &out_len, data, len) <= 0)
